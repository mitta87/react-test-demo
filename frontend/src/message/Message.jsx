import React from "react";
import "./Message.less";

const Message = (props) =>
	<div className="Message">
		<div className="Message__text">{props.message}</div>
		<div className="Message__buttonsPanel">
			<button className="Message__button" onClick={props.onLoadMessage}>Load message</button>
			<button className="Message__button" onClick={props.onDeleteMessage}>Delete message</button>
		</div>
	</div>;

export default Message;
