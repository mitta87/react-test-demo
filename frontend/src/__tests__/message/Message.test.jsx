import { shallow } from "enzyme";
import React from "react";
import Message from "../../message/Message";

describe("Message", () => {
	it("should render message from property", () => {
		//GIVEN
		const message = "Some message";
		const wrapper = shallow(<Message message={message}/>);

		//WHEN
		const actualMessage = wrapper.find(".Message__text").text();

		//THEN
		expect(actualMessage).toEqual(message);
	});

	it("should call callback function from props on 'load message' button click", () => {
		//GIVEN
		const onLoadMessage = jest.fn();
		const wrapper = shallow(<Message onLoadMessage={onLoadMessage}/>);

		//WHEN
		wrapper.find(".Message__button--load").simulate("click");

		//THEN
		expect(onLoadMessage.mock.calls.length).toEqual(1);
	});

	it("should call callback function from props on 'delete message' button click", () => {
		//GIVEN
		const onDeleteMessage = jest.fn();
		const wrapper = shallow(<Message onDeleteMessage={onDeleteMessage}/>);

		//WHEN
		wrapper.find(".Message__button--delete").simulate("click");

		//THEN
		expect(onDeleteMessage.mock.calls.length).toEqual(1);
	});

	it("should match snapshot", () => {
		//GIVEN
		const message = "Some message";

		//WHEN
		const wrapper = shallow(<Message message={message}/>);

		//THEN
		expect(wrapper).toMatchSnapshot();
	});
});