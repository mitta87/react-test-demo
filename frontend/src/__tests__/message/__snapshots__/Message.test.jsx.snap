// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should match snapshot 1`] = `
<div
  className="Message"
>
  <div
    className="Message__text"
  >
    Some message
  </div>
  <div
    className="Message__buttonsPanel"
  >
    <button
      className="Message__button Message__button--load"
    >
      Load message
    </button>
    <button
      className="Message__button Message__button--delete"
    >
      Delete message
    </button>
  </div>
</div>
`;
